modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/odata/ExpressionVisitorImpl.java

```diff
@@ -348,6 +384,45 @@ private String _getPropertyName(Object object) {
 		return String.valueOf(object);
 	}
 
+	private Object _getSub(Object left, Object right)
+		throws ExpressionVisitException {
+
+		if ((left instanceof String) && (right instanceof Duration)) {
+			try {
+				Date date = null;
+
+				if (Objects.equals(
+						MethodExpression.Type.NOW.toString(), left)) {
+
+					date = new Date();
+				}
+				else {
+					date = ISO8601Utils.parse(
+						String.valueOf(left), new ParsePosition(0));
+				}
+
+				Duration duration = Duration.parse(right.toString());
+
+				Instant instant = date.toInstant();
+
+				instant = instant.minusMillis(duration.toMillis());
+
+				return ISO8601Utils.format(Date.from(instant));
+			}
+			catch (ParseException parseException) {
+				throw new ExpressionVisitException(
+					"Invalid date format, use ISO 8601: " +
```

Comment:
Better if the message is something like only dates are supported in the Sub operation.

Reply:
ok, thanks